### ISSUE TITLE
container-independent access

### ISSUE THREAD

**Issue by KrisThielemans** [microsoft/yardl#21]
At present, I believe the user has to know if the stored data is binary or HDF5, and instantiate the corresponding class. That's efficient but also very inconvenient. It would certainly be nice to be able to write some client-code that does not depend on the container-type.

**Comment by johnstairs** [comment]
Yes, the readers and writers inherit from common abstract classes so once instantiated, you can have code that works independent of the file format. 

But we could generate factories. For readers, we could use the file extension or peek at the first few bytes. In addition to file names, the binary readers and writers support stream-like objects, but HDF5 does not support this, so I guess the factory would be based on a file name. For creating writers, you need to know what format you want, so there is probably less value in generating those.
